### DISCUSSION TITLE
built-in authentication with both username or client id

### DISCUSSION THREAD

**Discussion by simpsons310** [emqx/emqx#9636] [Q&A]
I'm using EMQX version 5.0.12, and this is my authentication config
```
authentication [
    "0" = {
        enable: false
        mechanism: password_based
        backend: built_in_database
        user_id_type: username
    }
]
```

However, I wonder there is a way to set `user_id_type` using `username` **OR** `client_id` with user. For example, user A can authenticate by `username` but user B also can authenticate by `client_id`

**Comment by lafirest** [comment]
@simpsons310 
I think you could simply duplicate the config and insert it to a listener authenticators list with change the `user_id_type` to `username`
See [Authentication of listener](https://www.emqx.io/docs/en/v5.0/security/authn/authn.html#authentication-of-listener) for details

**Reply by simpsons310** [reply]
@lafirest Sorry for late reply. I've tried to add new authentication via listener but I cannot check authentication configuration (setting and user list) via dashboard.

![image](https://user-images.githubusercontent.com/34018204/217510862-70add17c-85c6-47f5-8bd6-dca9ba29bdcf.png)

- Built-in database setting in dashboard has only one rule
![image](https://user-images.githubusercontent.com/34018204/217511997-d273da33-49f9-40f7-9039-58581dddd475.png)

- And rule `user_id_type` is set to `username`
![image](https://user-images.githubusercontent.com/34018204/217511797-0a9be8d3-3ad4-4ac7-9055-f9d434f73171.png)

This is my configuration:
```
listeners.tcp.default {
  bind = "0.0.0.0:1883"
  max_connections = 1024000
  authentication = [
    {
      enable = true
      mechanism = password_based
      backend = built_in_database
      user_id_type = clientid
    }
  ]
}

authentication = [
  {
    enable = true
    mechanism = password_based
    backend = built_in_database
    user_id_type = username
  },
  {
    enable = true
    mechanism = password_based
    backend = mysql
    query = "SELECT mqtt_password AS password_hash FROM ac_controllers WHERE mqtt_username = ${username} LIMIT 1"
    password_hash_algorithm {
      name = sha256
      salt_position = disable
    }
  }
]
```

**Comment by lafirest** [comment]
@simpsons310 
Sorry for the late reply too.
Now the dashboard can't display all configuration items of the listener, please wait for a while
